src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/ApplicationTest.kt

```diff
@@ -1872,295 +1874,426 @@ class ApplicationTest : IntegrationTestBase() {
+        withCrn(offenderDetails.otherIds.crn)
+        withApplicationSchema(
+          approvedPremisesApplicationJsonSchemaEntityFactory.produceAndPersist {
+            withDefaults()
+          },
+        )
+        withCreatedByUser(submittingUser)
+      }.id
+
+      CommunityAPI_mockSuccessfulRegistrationsCall(
+        offenderDetails.otherIds.crn,
+        Registrations(
+          registrations = listOf(
+            RegistrationClientResponseFactory()
+              .withType(RegistrationKeyValue(code = "MAPP", description = "MAPPA"))
+              .withRegisterCategory(RegistrationKeyValue(code = "M2", description = "M2"))
+              .withRegisterLevel(RegistrationKeyValue(code = "M2", description = "M2"))
+              .withStartDate(LocalDate.parse("2022-09-06"))
+              .produce(),
+          ),
+        ),
+      )
 
-            emailAsserter.assertEmailsRequestedCount(2)
-            emailAsserter.assertEmailRequested(
-              toEmailAddress = createdAssessment.allocatedToUser!!.email!!,
-              templateId = notifyConfig.templates.assessmentAllocated,
-              replyToEmailId = persistedApplication.apArea!!.notifyReplyToEmailId,
-            )
-            emailAsserter.assertEmailRequested(
-              toEmailAddress = submittingUser.email!!,
-              templateId = notifyConfig.templates.applicationSubmitted,
-              replyToEmailId = persistedApplication.apArea!!.notifyReplyToEmailId,
-            )
-          }
-        }
-      }
+      APDeliusContext_mockSuccessfulCaseDetailCall(
+        offenderDetails.otherIds.crn,
+        CaseDetailFactory().produce(),
+      )
+
+      GovUKBankHolidaysAPI_mockSuccessfullCallWithEmptyResponse()
+
+      val overriddenApArea = `Given an AP Area`(name = "wales")
+
+      webTestClient.post()
+        .uri("/applications/$applicationId/submission")
+        .header("Authorization", "Bearer $jwt")
+        .bodyValue(
+          SubmitApprovedPremisesApplication(
+            noticeType = Cas1ApplicationTimelinessCategory.shortNotice,
+            apType = ApType.pipe,
+            translatedDocument = {},
+            isWomensApplication = false,
+            targetLocation = "SW1A 1AA",
+            releaseType = ReleaseTypeOption.licence,
+            sentenceType = SentenceTypeOption.nonStatutory,
+            type = "CAS1",
+            apAreaId = overriddenApArea.id,
+            applicantUserDetails = Cas1ApplicationUserDetails("applicantName", "applicantEmail", "applicationPhone"),
+            caseManagerIsNotApplicant = false,
+          ),
+        )
+        .exchange()
+        .expectStatus()
+        .isOk
+
+      val persistedApplication = approvedPremisesApplicationRepository.findByIdOrNull(applicationId)!!
+
+      assertThat(persistedApplication.noticeType).isEqualTo(Cas1ApplicationTimelinessCategory.shortNotice)
+      assertThat(persistedApplication.apArea?.id).isEqualTo(overriddenApArea.id)
+
```

Comment:
Observation: this test doesn't seem to test domain events for this scenario.

Reply:
for consistency all tests are checking domain events and i've updated the test names to reflect this